### PR TITLE
added branch type

### PR DESCRIPTION
**Pull request by mineshb-transerve**
pe - 68 - added branch type

### CODE REVIEW

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java [hunk 1]
```diff
@@ -61,10 +59,38 @@ public void validateCreateRequest(PropertyRequest request) {
...
+				roleCode.add(r.getCode());
+			});
+		}
+		
+		
+		//check with proerpty have branch type as per userinfo role
+		List<Property> propertyList = request.getProperties();
+		for (Property property_ : propertyList) {
+			if(null != property_.getPropertyDetails() && 
+					!roleList.contains(property_.getPropertyDetails().getBranchType())){
```
Comment (rajiv-transerve) [inline property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:86]: Role will be like `EB_APPROVER` but the branch type will be `EstateBranch`. Both will not be the same.

**Review by rajiv-transerve — CHANGES_REQUESTED** [review]

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java [hunk 2]
```diff
@@ -61,10 +59,38 @@ public void validateCreateRequest(PropertyRequest request) {
...
+	private void validateRole(PropertyRequest request, Map<String, String> errorMap) {
+		
+		//fetch all userinfo roles...
+		RequestInfo requestInfo = request.getRequestInfo();
+		List<org.egov.common.contract.request.Role> roleList = null;
+		if(null != requestInfo.getUserInfo()) {
+			roleList = requestInfo.getUserInfo().getRoles();
+			List<String> roleCode =  new ArrayList<String>(0);
+					
+			roleList.stream().forEach(r -> {
```
[the remark this answers is not in the document]
Comment (mineshb-transerve) [reply property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:76]: Yes that true but i don't have Role class access this line so i used List<String> with using for each...

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java [hunk 3]
```diff
@@ -61,10 +59,38 @@ public void validateCreateRequest(PropertyRequest request) {
 
 		Map<String, String> errorMap = new HashMap<>();
 
+		validateRole(request, errorMap);
```
[the remark this answers is not in the document]
Comment (mineshb-transerve) [reply property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:62]: I dont getting this point ...is MDMS have any parent json file which provide details list of role .

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java (hunk 1 above)
Comment (mineshb-transerve) [reply property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:86]: Ok Agree but need to discussion ... currently in request we have branch type and role have EB_APPROVE ...s o maching base on string matches ?

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajiv-transerve) [reply property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:77]: I did not understand this.

File: property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java [hunk 4]
```diff
@@ -61,10 +60,49 @@ public void validateCreateRequest(PropertyRequest request) {
...
+		}
+		
+		//fetch all mdms data for branch type 
+		List<Map<String, Object>> fieldConfigurations = mdmsservice.getBranchRoles("branchtype", request.getRequestInfo(), request.getProperties().get(0).getTenantId());
+		System.out.println(fieldConfigurations);
+		ObjectMapper mapper = new ObjectMapper();
+		List<Role> roleListMdMS = mapper.convertValue(fieldConfigurations, new TypeReference<List<Role>>() { });
+		
+		// check with user role is present...
+		List<Property> propertyList = request.getProperties();
```
Comment (rajiv-transerve) [inline property-service/src/main/java/org/egov/ps/validator/PropertyValidator.java:89]: This can be rewritten using streams..

**Review by rajiv-transerve — CHANGES_REQUESTED** [review]

**Review by rajiv-transerve — APPROVED** [review]

**Pull request merged**